app/src/main/java/com/lagradost/cloudstream3/ui/player/AbstractPlayerFragment.kt

```diff
@@ -607,23 +617,57 @@ abstract class AbstractPlayerFragment(
     }
 
     fun nextResize() {
-        resizeMode = (resizeMode + 1) % PlayerResize.entries.size
-        resize(resizeMode, true)
+        showAspectRatioDialog()
     }
 
     fun resize(resize: Int, showToast: Boolean) {
         resize(PlayerResize.entries[resize], showToast)
     }
 
+    private fun showAspectRatioDialog() {
+        player.handleEvent(CSPlayerEvent.Pause, PlayerEventSource.UI)
+
+        val aspectRatioOptions = PlayerResize.entries.map { getString(it.nameRes) }
+
+        activity?.showDialog(
+            aspectRatioOptions,
+            DataStoreHelper.resizeMode,
+            getString(R.string.choose_aspect_ratio),
+            false, // Immediate selection without an Apply button
+            dismissCallback = {
+                player.handleEvent(CSPlayerEvent.Play, PlayerEventSource.UI)
+            },
+            callback = { selectedIndex ->
+                val selectedOption = PlayerResize.entries[selectedIndex]
+                resize(selectedOption, true)
+                player.handleEvent(CSPlayerEvent.Play, PlayerEventSource.UI)
+            }
+        )
+    }
+
+
+
     @SuppressLint("UnsafeOptInUsageError")
     fun resize(resize: PlayerResize, showToast: Boolean) {
         DataStoreHelper.resizeMode = resize.ordinal
         val type = when (resize) {
             PlayerResize.Fill -> AspectRatioFrameLayout.RESIZE_MODE_FILL
             PlayerResize.Fit -> AspectRatioFrameLayout.RESIZE_MODE_FIT
             PlayerResize.Zoom -> AspectRatioFrameLayout.RESIZE_MODE_ZOOM
+            PlayerResize.SixteenByNine -> 16.0f/9.0f
+            PlayerResize.NineBySixteen -> 9.0f/16.0f
+            PlayerResize.OneByOne -> 1.0f/1.0f
+            PlayerResize.IMAX -> 1.43f/1.0f
+            PlayerResize.DolbyVision -> 2.39f/1.0f
+            PlayerResize.UltraWide -> 21.0f/9.0f
+            PlayerResize.Expanded -> 1.90f/1.0f
+        }
+        if(type is Float){
+            playerView?.resizeMode = AspectRatioFrameLayout.RESIZE_MODE_FIT
+            aspectRatioFrameLayout?.setAspectRatio(type)
+        }else if(type is Int) {
+            playerView?.resizeMode = type
         }
-        playerView?.resizeMode = type
 
         if (showToast)
             showToast(resize.nameRes, Toast.LENGTH_SHORT)
```

Comment:
Since after this change the aspect ratio is explicitly set by the user, it may be worth thinking if this toast is necessary.

Reply:
Removed Toast from the Resize function.